### ISSUE TITLE
How can classification models be used for images as small as 16x16? I could use some help

### ISSUE THREAD

**Issue by leyestd** [ultralytics/ultralytics#21071]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

[](url)

![Image](https://github.com/user-attachments/assets/b6df6991-80ef-4c4f-8bdf-ba836266bec5)
![Image](https://github.com/user-attachments/assets/7787491e-9ff7-4774-a848-fe7e8c270dde)   
The images are very small, what's the best way to process them? Thanks.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @leyestd, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, where many common questions are already addressed.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If this is a classification or custom training ❓ Question, please include as much detail as possible—such as more image examples, information about your dataset, model settings, and any relevant training logs. Also, be sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimizing your workflow.

Connect with the Ultralytics community for more support:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn with others

## Upgrade

Upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue is not already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will review your question and assist you soon! 😊

**Comment by Y-T-G** [comment]
The minimum Ultralytics support is 32x32. The images will be automatically resized to that size.

**Comment by leyestd** [comment]
Thanks. What methods or parameters are best for detecting defects in a mix of 16x16, 20x20, and 50x50 images? I mainly need this to distinguish between different defect categories. Could you teach me the optimal parameter configuration, or what parts I should modify?

![Image](https://github.com/user-attachments/assets/97b0f377-8925-4b3f-ab3d-dc957af1d938)

![Image](https://github.com/user-attachments/assets/d8188ca1-00db-4a2d-a5f1-765f7c67bb18)

**Comment by Y-T-G** [comment]
I think you can try the default parameters. Just set the dataset and epochs.

**Comment by goodevile** [comment]
> Thanks. What methods or parameters are best for detecting defects in a mix of 16x16, 20x20, and 50x50 images?

now, 16x16 is VERY small... I saw that your first images had 300x300.... can't you use 64x64?
the smallest is 32x32 and I don't recommend resizing it if there is a big difference.

360x360 to 352x352? No problem in 99% of the cases.
16x16 to 32x32? I, personally, woudn't do it for a quality control step in production.



obs.:
you are not providing sufficient information, can't help you with more than this.
hardware?
camera resolution?

BTW, even the famous raspberry pi 4 can do 64x64 in real time.

**Comment by leyestd** [comment]
I need to detect if there are problems with the small dots in a large image. My process is to first use object detection to crop out the small dots, and then perform classification. Some of the cropped images are as small as 16x16 pixels. These are very small holes on a PCB. Is there a way to improve the accuracy without changing the hardware?

**Comment by goodevile** [comment]
> I need to detect if there are problems with the small dots in a large image. My process is to first use object detection to crop out the small dots, and then perform classification. Some of the cropped images are as small as 16x16 pixels. These are very small holes on a PCB. Is there a way to improve the accuracy without changing the hardware?

I don't know because I don't know your hardware.
I am not trying to be a bad person here but you need to know how to provide information that is needed before asking questions.
No info = time wasted answering your question.

**Comment by glenn-jocher** [comment]
For your PCB defect classification workflow, since the 16x16 crops will be upsampled to 32x32 automatically, consider these approaches to improve accuracy: use data augmentation during training to help the model generalize better with the upsampled images, ensure your training dataset includes similar quality crops at the same resolution, and experiment with different classification model sizes - sometimes smaller models like yolo11n-cls perform better on constrained data than larger ones. You might also try cropping slightly larger regions (24x24 or 32x32) during detection if possible, as this could preserve more defect details before classification.

**Comment by leyestd** [comment]
Thanks everyone for the great suggestions! We'll definitely look into these.

**Comment by glenn-jocher** [comment]
You're welcome! Feel free to reach out if you run into any specific issues during implementation. Good luck with your PCB defect detection project!
